### ISSUE TITLE
Issue with shared datasets

### ISSUE THREAD

**Issue by gmuenster** [microsoft/RdlMigration#26]
Hi,

I have a report using a shared data source and two shared datasets. The datasets are referring to the same shared data source. The report can be executed successfully, but when I try to migrate it, I always get the error message: "Can't find corresponding Data Set". Is there something specific I need to consider?

Thanks in advance.

**Comment by clarkxde** [comment]
Hi,
all my report conversions get this message "FAILED : DataDump_SODate Can't find corresponding Data Set dsDataDump" where the datasets are in the subfolder "Datasets". According to the documentation this should work, or am I also missing something?

Thank you

**Comment by gboreki** [comment]
Yes, this should work. Do you mind showing the path of the report and of the shared dataset? Was the shared dataset reference changed in the server (after the report was design in the client tool)? And also what is your SSRS version and mode (native/sharepoint)?

**Comment by clarkxde** [comment]
Hi @Gboreki,
This is the command I ran:
RdlMigration https://reports.mycompany.local/reportserver "/Sales" "Sales" 999999-xxxx-xxxx-xxxx-99999999

Log-in successfully, retrieving the reports...
Starting conversion and uploading the reports 6/18/2020 9:07:56 AM
FAILED : Sales_Aggregated Can't find corresponding Data Set dsSalesAggregatedDD

The structure is different to default, but I thought this may be an issue, so I actually redeployed my reports to a different folder using the default folder structure, but that did not help, still the same error. I saw the first step downloaded all the reports locally, but not the datasource nor dataset, if I put them in the folder manually, would that help?
![ReportStructure](https://user-images.githubusercontent.com/53169810/85113036-f05ef100-b216-11ea-94fd-f9bea064ec52.GIF)
![SSRSVersion](https://user-images.githubusercontent.com/53169810/85113057-f8b72c00-b216-11ea-9e6f-e7ccfb7afa68.GIF)

**Comment by vishalbiggest** [comment]
We are facing the same issue with Shared DataSets. Shared Data Source works.

**Comment by gboreki** [comment]
I have submitted a pull request that addresses an issue with shared dataset mapping. Would some of you be able to validate if that is the issue that is affecting you? I have validated the change against a native mode PBIRS instance and I am looking for a sharepoint integrated mode to check as well.

This is the release from PR: https://github.com/microsoft/RdlMigration/archive/CI_2110572.zip
And this is the PR: https://github.com/microsoft/RdlMigration/pull/27

**Comment by clarkxde** [comment]
Hi @gboreki gboreki,

I tried with you updated version and now I get the following message on all reports - i.e. none get converted:

Starting the log-in window
Log-in successfully, retrieving the reports...
Starting conversion and uploading the reports 7/28/2020 8:17:13 AM
FAILED TO UPLOAD :  Sales RequestId:9....9 Report is invalid. ErrorCode: rsDuplicateDataSourceName, Details: More than one data source in the report has the name DataSource0_Securitization. Data source names must be unique within a report.

**Comment by nikejsf** [comment]
Hi @clarkxde, did you ever figure out your issue? i'm having the same problem with all my reports saying there is more than one data source named the same thing. Much appreciated! Thanks

> Hi @gboreki gboreki,
> 
> I tried with your updated version and now I get the following message on all reports - i.e. none get converted:
> 
> Starting the log-in window
> Log-in successfully, retrieving the reports...
> Starting conversion and uploading the reports 7/28/2020 8:17:13 AM
> FAILED TO UPLOAD : Sales RequestId:9....9 Report is invalid. ErrorCode: rsDuplicateDataSourceName, Details: More than one data source in the report has the name DataSource0_Sales. Data source names must be unique within a report.

**Comment by clarkxde** [comment]
> Hi @clarkxde, did you ever figure out your issue? i'm having the same problem with all my reports saying there is more than one data source named the same thing. Much appreciated! Thanks
> 
Hi @nikejsf  - Sorry, never heard back from @gboreki - so I have not persued it further, assume I will have to do a lot of manual work to get these migrated. Wonder why @Microsoft advises the use of this tool when it is not properly supported nor fully functional ...

**Comment by KRK-Bowman** [comment]
Could we please get a release that contains this fix?

**Comment by BasileGruel** [comment]
Hello, I tried various versions but didn't find a working branch in GIT. From the branch of this Fixed bug PR (that clear out duplicates) you end up with a discrepancy with Datasource0_XXXXX in the Datasource and Datasource1_XXXXX in the Dataset. Just getting rid of the digits in DatasourceName via regex seems to be working but I couldn't run extensive tests (c# is not my domain of expertise I just need those reports to move)
